lib/role.rb

```diff
@@ -4,11 +4,17 @@
 # convenience methods for testing the role status of users.
 module Role
 
+  def grant_role!(role, facility: nil)
```

Comment:
Yep I missed taking this out. I used it in a WIP that I ended up rebasing away. I do like the interface of this method more: `user.grant_role!(:facility_staff, facility: facility)` beats `UserRole.grant(user, FACILITY_STAFF, facility)` for me.

`UserRole.grant` mostly seems like a way to write `UserRole.create!` using fewer characters, and as you mentioned doesn't do anything to prevent duplicate role assignments. It looks like `UserRole.grant` is barely used outside specs and seeds anyway. I'm inclined to throw out both methods and add a validation.